### PR TITLE
Feature event gallery implemented

### PR DESCRIPTION
**Pull request by bremerjung**
vertical slice for event list view
setup database and import example events from backend/src/test/resources/events.json
use Link "Zu den Events" on Home (http://localhost:3000/) to get to event gallery view

### CODE REVIEW

**Review by Kreyno93 — COMMENTED** [review]
recommendation: 
work with vertical slices
over 900 lines of code, which could be chopped into smaller pieces
get,post,put, delete -> all in one slice
its much better to slice it up so the review doesnt take that long and the re roll is much easier if its broken after merge.
and u have to make sure ALL feature are working instead of just one.

File: backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                                        "description": "Description",
+                                        "start": "2023-06-14T10:00:00Z",
+                                        "end": "2023-06-14T12:00:00Z",
+                                        "location": "Location",
+                                        "category": "MUSIC",
+                                        "creator": "Creator",
+                                        "status": "NEW"
+                                }
+                                """))
+                .andExpect(status().is(201));
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java:42]: u are returning the event object as well, calling the method. thats missing in the expect part of the test.

File: backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                                        "description": "Description",
+                                        "start": "2023-06-14T10:00:00Z",
+                                        "end": "2023-06-14T12:00:00Z",
+                                        "location": "Location",
+                                        "category": "MUSIC",
+                                        "creator": "Creator",
+                                        "status": "NEW"
+                                }
+                                """))
+                .andExpect(status().is(201))
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java:88]: same

File: backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java:130]: same

File: backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java:188]: same

File: backend/src/main/java/de/neuefische/backend/dto/EventDTO.java
Comment (chris-diesner) [inline backend/src/main/java/de/neuefische/backend/dto/EventDTO.java:1]: Does it make sense to have an identical DTO for the equivalent model?

**Review by Kreyno93 — APPROVED** [review]

File: backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java (hunk 1 above)
Comment (chris-diesner) [inline backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java:42]: ...andExpect for returning object?

File: backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                                        "description": "Updated description",
+                                        "start": "2023-06-15T10:00:00Z",
+                                        "end": "2023-06-15T12:00:00Z",
+                                        "location": "New Location",
+                                        "category": "THEATRE",
+                                        "creator": "Creator 2",
+                                        "status": "APPROVED"
+                                }
+                                """))
+                .andExpect(status().isOk());
```
Comment (chris-diesner) [inline backend/src/test/java/de/neuefische/backend/controller/EventControllerTest.java:153]: ^^

**Review by chris-diesner — CHANGES_REQUESTED** [review]

File: backend/src/main/java/de/neuefische/backend/dto/EventDTO.java
Comment (bremerjung) [reply backend/src/main/java/de/neuefische/backend/dto/EventDTO.java:1]: Unfortunately, this was unavoidable. When using a persistent entity as an argument of an "@RequestMapping" method, SonarCloud raises a critical security vulnerability: Persistent entities should not be used as arguments of "@RequestMapping" methods (see https://rules.sonarsource.com/java/RSPEC-4684).

Comment (chris-diesner) [reply backend/src/main/java/de/neuefische/backend/dto/EventDTO.java:1]: aaaah...good to know!

**Review by chris-diesner — APPROVED** [review]
nice work so far!

**Pull request merged**